### ISSUE TITLE
Assembly-wide ExcludeFromCodeCoverageAttribute is ignored

### ISSUE THREAD

**Issue by Nezz** [SonarSource/sonar-dotnet#7533]
### Description

When adding the `ExcludeFromCodeCoverageAttribute` to a whole assembly, the assembly does not get excluded from code coverage metrics.

### Repro steps
Create an `AssemblyInfo.cs` in a C# project with the following contents:
```
using System.Diagnostics.CodeAnalysis;

[assembly:ExcludeFromCodeCoverage]
```

### Expected behavior

The code should be excluded from coverage.

### Actual behavior

The code is reported with 0% coverage because the coverage tool excludes it already.

### Known workarounds

Manual exclusion through other means.

### Related information

Checking the source code it seems that this case is not handled:
https://github.com/SonarSource/sonar-dotnet/blob/cafe6bbf511e33b5eb976bf4a29637a4b2c5d9e0/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Metrics/ExecutableLinesWalker.cs#L96-L114

**Comment by gregory-paidis-sonarsource** [comment]
Hey @Nezz!

Thanks for raising this. 
I can confirm that the `ExecutableLinesWalker` indeed does not take account of an assembly-wide exclusion from code coverage. 

I added it to our backlog, where it will be taken care of by some upcoming sprint in the future.
Cheers! :)

**Comment by pavel-mikula-sonarsource** [comment]
Not generating coverage for a specific project seems like a task for the coverage tool and not for us.

Also `ExecutableLinesWalker` doesn't sound like a think that should affect coverage.

**Comment by gregory-paidis-sonarsource** [comment]
@pavel-mikula-sonarsource that's a good point. Do you think we should close this?

**Comment by pavel-mikula-sonarsource** [comment]
Yes. It will also improve the UT performance in CI when the coverage tool is not executed.

We exclude projects from coverage reporting like this:
https://github.com/SonarSource/sonar-dotnet/blob/6bea9f4cc50bd86cbe6bfd562cf75119cbd686cb/azure-pipelines.yml#L176
